sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/response/CreateUserResponse.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.rest.response;
+
+import org.opensearch.action.ActionResponse;
+import org.opensearch.common.ParseField;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ConstructingObjectParser;
+import org.opensearch.common.xcontent.StatusToXContentObject;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.opensearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.opensearch.rest.RestStatus.NOT_FOUND;
+import static org.opensearch.rest.RestStatus.OK;
+
+/**
+ * Response class for create user flow
+ */
+public class CreateUserResponse extends ActionResponse implements StatusToXContentObject {
+
+    // TODO: revisit this class
+    private final List<CreateUserResponseInfo> createUserResults;
+
+    public CreateUserResponse(List<CreateUserResponseInfo> createUserResults) {
+        this.createUserResults = createUserResults;
+    }
+
+    public CreateUserResponse(StreamInput in) throws IOException {
+        super(in);
+        int size = in.readVInt();
+        createUserResults = new ArrayList<>();
+        for (int i = 0; i < size; i++) {
+            createUserResults.add(new CreateUserResponseInfo(in));
+        }
+
+    }
+
+    public List<CreateUserResponseInfo> getCreateUserResults() {
+        return createUserResults;
+    }
+
+    /**
+     * @return Whether the attempt to Create a user was successful
+     */
+    @Override
+    public RestStatus status() {
+        if (createUserResults.isEmpty()) return NOT_FOUND;
+        return OK;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeVInt(createUserResults.size());
```

Comment:
Is that idea here to support a batched version of this API in the future?

Reply:
I kept this thinking we might support to create multiple users at a time. If not needed I shall remove it